recipes/otfftpp/all/conanfile.py

```diff
@@ -0,0 +1,116 @@
+from conan import ConanFile
+from conan.tools.cmake import CMakeToolchain, CMake, cmake_layout, CMakeDeps
+from conan.tools.files import copy, get
+import os
+
+class otfftppRecipe(ConanFile):
+    name = "otfftpp"
+    version = "0.0.2"
+    package_type = "library"
+    license = "MIT"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/robinchrist/otfftpp"
+    description = "OTFFT is a high-speed FFT library using the Stockham's algorithm and SIMD"
+    topics = ("FFT", "SIMD")
+
+    settings = "os", "compiler", "build_type", "arch"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "abi_affecting_cflags": ["ANY"], #comma separated list of c-flags, will be converted to a set for normalization and sorting
+        "abi_affecting_cxxflags": ["ANY"], #comma separated list of c-flags, will be converted to a set for normalization and sorting
```

Comment:
Do you have an example of flags that you would pass via these options? Something like `-march=...`?
Wouldn't passing these flags globally via `tools.build:cflags=...` and `tools.build:cxxflags=...` profile options be sufficient?

Reply:
Yes exactly, march.
As an example from my testing profile:
```
otfftpp/*:abi_affecting_cflags=-march=x86-64-v3,-mtune=znver3
otfftpp/*:abi_affecting_cxxflags=-march=x86-64-v3,-mtune=znver3
```

The decision to do it that way has been the result of careful thinking, for my thoughts see here: https://github.com/conan-io/conan/issues/847#issuecomment-2524181706

From my point of view, it is ESSENTIAL that the flags specified change the package id. Otherwise you can end up with packages in your local cache that do not match your expected flags (I think this has also been discussed in https://github.com/conan-io/conan/issues/847)

https://github.com/conan-io/conan-center-index/tree/master/recipes/unicorn uses a similar approach

Reply:
Re package id: Specifically this https://github.com/conan-io/conan/issues/847#issuecomment-364060269

As far as I'm aware, setting `tools.build:cflags=...`  wouldn't change the package id, which is very bad in case of a high performance FFT library (you might end up getting a SSE version and not the AVX version you want)

Reply:
Thanks! I'm a bit surprised that the global config vars are not taken into account in the package ID by Conan already...

Reply:
I just verified this:
First, delete stuff from cache:

`conan remove -c otfftpp`

A first run of `conan install . --output-folder build --build=missing <profile>` of a project that consumes this package with profile specifying 
```
tools.build:cflags =+ ["-march=x86-64-v3", "-mtune=znver3"]
tools.build:cxxflags =+ ["-march=x86-64-v3", "-mtune=znver3"]
```
(no abi_affecting_cxxflags specified)
-> First run compiled the package from source, building & running the consuming application that links against the library will give:
```
  otfftpp: __AVX__ 1
  otfftpp: __AVX2__ 1
  Main App: __AVX__ 1
  Main App: __AVX2__ 1
```

Now change profile to (and delete build folder to be sure):
```
tools.build:cflags =+ ["-march=x86-64-v2", "-mtune=generic"]
tools.build:cxxflags =+ ["-march=x86-64-v2", "-mtune=generic"]
```
Second run of `conan install . --output-folder build --build=missing <profile>` does not recompile otfftpp (already in cache)
building & running the consuming application that links against the library will give:
```
  otfftpp: __AVX__ 1
  otfftpp: __AVX2__ 1
  Main App: __AVX__ 0
  Main App: __AVX2__ 0
```

So even though I wanted the entire app to be built WITHOUT AVX (x86-64-v2 has no AVX), otfftpp is still using AVX as it is in the cache and the global compile flags don't affect the package id (and it has before been built with x86-64-v3 -> AVX enabled)



This is fixed if I explicitly set the flags via `otfftpp/*:abi_affecting_cflags` as this changes the package id.
I think this would be the first package to do it that way - However there's always a first :)